SwiftReflector/VTableDetails.cs

```diff
@@ -0,0 +1,490 @@
+
+
+			currentVTEntry++;
+			return recvr;
+		}
+
+		void ImplementUnmanagedReceiver (CSMethodKind methodKind,CSDelegateTypeDecl delType,
+			CSParameterList pl, CSParameterList plWithoutGCPtr, List<string> usedIDs, CSUsingPackages use)
+		{
+			var body = new CSCodeBlock ();
+
+			var unmanagedParameterList = new CSParameterList ();
+			var args = new List<CSBaseExpression> ();
+			unmanagedParameterList.Add (pl [0]); // add in the gc handle
+
+			// the compiler doesn't like delegate *unmanaged<SomeType *>
+			// why? Couldn't say, so we rewrite any pointer types as
+			// IntPtr and then case them as the original pointer
+
+			for (var i = 1; i < pl.Count; i++) {
+				var parameter = pl [i];
+				var theType = parameter.CSType;
+				var maybePtr = ToMaybeIntPtr (theType);
+
+				if (maybePtr == theType) {
+					unmanagedParameterList.Add (parameter);
+					args.Add (parameter.Name);
+				} else {
+					unmanagedParameterList.Add (new CSParameter (maybePtr, parameter.Name, parameter.ParameterKind));
+					args.Add (new CSCastExpression (theType, parameter.Name));
+				}
+			}
+
+			var unmanagedReceiver = new CSMethod (CSVisibility.None, methodKind, delType.Type,
+				new CSIdentifier (RecevierName ()), unmanagedParameterList, body);
+			var gchName = NewClassCompiler.Uniqueify ("gch", usedIDs);
+			usedIDs.Add (gchName);
+			var gchID = new CSIdentifier (gchName);
+			body.Add (CSVariableDeclaration.VarLine (gchID, new CSFunctionCall ("GCHandle.FromIntPtr", false, pl [0].Name)));
+			var localVTName = NewClassCompiler.Uniqueify ("localVT", usedIDs);
+			usedIDs.Add (localVTName);
+			var localVTID = new CSIdentifier (localVTName);
+			body.Add (CSVariableDeclaration.VarLine (localVTID, new CSCastExpression (LocalCSVTable.ToCSType (),
+				new CSParenthesisExpression (CSUnaryExpression.PostBang (gchID.Dot (new CSIdentifier ("Target")))))));
+
+			var invocation = new CSFunctionCall ($"{localVTName}.{FuncName ()}!", false, args.ToArray ());
+
+			var callLine = delType.Type == CSSimpleType.Void ? new CSLine (invocation) :
+				CSReturn.ReturnLine (invocation);
+			body.Add (callLine);
+			use.AddIfNotPresent (typeof (UnmanagedCallersOnlyAttribute));
+			var attr = CSAttribute.FromAttr (typeof (UnmanagedCallersOnlyAttribute), new CSArgumentList (), true);
+			attr.AttachBefore (unmanagedReceiver);
+			UnmanagedReceivers.Methods.Add (unmanagedReceiver);
+		}
+
+		public CSMethod ImplementVirtualPropertyStaticReceiver (CSType thisType, string csProxyName, CSDelegateTypeDecl delType,
+					       CSUsingPackages use, FunctionDeclaration funcDecl, CSProperty prop, CSMethod protoListMethod,
+					       TypeMapper typeMapper, bool isObjC, bool hasAssociatedTypes, Func<int, int, string> genericRenamer = null)
+		{
+			var returnType = funcDecl.IsGetter ? delType.Type : CSSimpleType.Void;
+
+			var pl = delType.Parameters;
+			var plWithoutGCPtr = new CSParameterList (pl.GetRange (1, pl.Count - 1));
```

Comment:
Careful, I don't know if possible but if p1.Count is 0, GetRange will throw an exception: https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.getrange?view=net-8.0#exceptions

Reply:
All the parameter lists that make to this point will have the GC Handle as the first argument. Although I think I'm going to change that to `Skip (1)` since it reads better.